### PR TITLE
#356 added endpoint to fetch all cluster connectors

### LINKED ISSUE #356: Add endpoint to fetch all cluster connectors
/clusters/:clusterName/connectors

Connector:
- connect: string
- name: string
- class/plugin
- type: source | sync
- topics: [topicName]
- status: failed | running?
- tasks_count: int
- hasFailedTasks: bool

### PR DESCRIPTION
**Pull request by RamazanYapparov**
Fixes #356

### CODE REVIEW

**Review by workshur — APPROVED** [review]
LGTM

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 1]
```diff
@@ -1709,3 +1732,31 @@ components:
...
+            - source
+            - sink
+        topics:
+          type: array
+          items:
+            type: string
+        status:
+          type: string
+          enum:
+            - RUNNING
```
[the remark this answers is not in the document]
Comment (RamazanYapparov) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml]: Fixed, thanks

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 2]
```diff
@@ -1709,3 +1732,31 @@ components:
...
+      type: object
+      properties:
+        connect:
+          type: string
+        name:
+          type: string
+        type:
+          type: string
+          enum:
+            - source
```
[the remark this answers is not in the document]
Comment (RamazanYapparov) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml]: Fixed, thanks

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaConnectService.java [hunk 3]
```diff
@@ -43,6 +53,58 @@
     );
   }
 
+  public Flux<FullConnectorInfo> getAllConnectors(String clusterName) {
+    return getConnects(clusterName)
```
[the remark this answers is not in the document]
Comment (RamazanYapparov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaConnectService.java:55]: Fixed, thanks

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaConnectService.java [hunk 4]
```diff
@@ -43,6 +53,58 @@
...
+        .map(triple -> new FullConnectorInfo()
+            .connect(triple.getLeft().getConnect())
+            .name(triple.getLeft().getName())
+            .type(FullConnectorInfo.TypeEnum.valueOf(triple.getLeft().getType().name()))
+            .topics(getTopicsFromConfig(triple.getMiddle()))
+            .status(
+                FullConnectorInfo.StatusEnum.valueOf(triple.getLeft().getStatus().getState().name())
+            )
+            .tasksCount(triple.getRight().size())
+            .hasFailedTasks(triple.getRight().stream()
```
[the remark this answers is not in the document]
Comment (RamazanYapparov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaConnectService.java]: Fixed, thanks

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaConnectService.java [hunk 5]
```diff
@@ -43,6 +53,58 @@
...
+        .flatMap(connector ->
+            getConnectorConfig(clusterName, connector.getConnect(), connector.getName())
+                .map(config -> Pair.of(connector, config))
+        )
+        .flatMap(pair ->
+            getConnectorTasks(clusterName, pair.getLeft().getConnect(), pair.getLeft().getName())
+                .collectList()
+                .map(tasks -> Triple.of(pair.getLeft(), pair.getRight(), tasks))
+        )
+        .map(triple -> new FullConnectorInfo()
```
[the remark this answers is not in the document]
Comment (RamazanYapparov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaConnectService.java]: Fixed, thanks

**Pull request merged**